lib/ActivityStreams.js

```diff
@@ -170,6 +176,23 @@ ActivityStreams.prototype = {
   },
 
   /**
+   * Instantiate the recommender that scores the highlights items.
+   * Called when weightedHighlights prefs is toggled to true.
+   * @private
+   */
+  _loadRecommender() {
+    // Only need to load history items once per session.
+    if (this._baselineRecommender !== null) {
+      return;
+    }
+
+    this._memoized.getAllHistoryItems().then(historyItems => {
+      let highlightsCoefficients = JSON.parse(simplePrefs.prefs.weightedHighlightsCoefficients);
```

Comment:
let's wrap this in a try/catch to handle faulty pref values and then make the constructor fallback to defaults